src/network/protocol/redis/command/network_protocol_redis_command_mget.c

```diff
@@ -0,0 +1,316 @@
+                            db,
+                            chunk_info,
+                            chunk_send_buffer);
+
+                    if (!res) {
+                        LOG_E(
+                                TAG,
+                                "[REDIS][MGET] Critical error, unable to read chunk <%u> long <%u> bytes",
+                                chunk_index,
+                                chunk_info->chunk_length);
+
+                        goto end;
+                    }
+
+                    if (network_send(
+                            channel,
+                            chunk_send_buffer,
+                            chunk_info->chunk_length) != NETWORK_OP_RESULT_OK) {
+                        goto end;
+                    }
+
+                    slab_allocator_mem_free(chunk_send_buffer);
+                }
+
+                // At this stage the entry index is not accessed further therefore the readers counter can be decreased. The
+                // entry_index has to be set to null to avoid that it's freed again at the end of the function
+                storage_db_entry_index_status_decrease_readers_counter(entry_index, NULL);
+                entry_index = NULL;
+
+                send_buffer_start = protocol_redis_writer_write_argument_blob_end(
+                        send_buffer_start,
+                        send_buffer_end - send_buffer_start);
+        } else {
+            if (protocol_context->resp_version == PROTOCOL_REDIS_RESP_VERSION_2) {
+                send_buffer_start = protocol_redis_writer_write_blob_string_null(
+                        send_buffer_start,
+                        send_buffer_end - send_buffer_start);
+            } else {
+                send_buffer_start = protocol_redis_writer_write_null(
+                        send_buffer_start,
+                        send_buffer_end - send_buffer_start);
+            }
+        }
+    }
+
+    if (network_send(
+            channel,
+            send_buffer,
+            send_buffer_start - send_buffer) != NETWORK_OP_RESULT_OK) {
+        goto end;
+    }
+
+    // return_res is set to false at the beginning and switched to true only at this stage, this helps to avoid code
+    // duplication for the cleanup
+    return_res = true;
+
+end:
+    if (entry_index != NULL) {
+        storage_db_entry_index_status_decrease_readers_counter(entry_index, NULL);
+    }
+
+    return return_res;
+}
+
+NETWORK_PROTOCOL_REDIS_COMMAND_FUNCPTR_COMMAND_FREE(mget) {
+    if (!protocol_context->command_context) {
+        return true;
+    }
+
+    mget_command_context_t *mget_command_context = (mget_command_context_t*)protocol_context->command_context;
+
+    if (mget_command_context->key_data == NULL) {
+        slab_allocator_mem_free(mget_command_context->key_data);
+    }
+
+    for(int key_index = 0; key_index < mget_command_context->keys_count; key_index++) {
+        slab_allocator_mem_free(mget_command_context->keys[key_index]);
```

Comment:
Potentially a key might be null because the mget command aborted before receiving all the keys, it's better to free the key only if the value is not null (I dropped a related command at the line where the memory is allocated)